sakura_core/CBackupAgent.cpp

```diff
@@ -433,11 +432,9 @@ bool CBackupAgent::FormatBackUpPath(
 				else {
 					ptr = szExt + wcslen( szExt );
 				}
-				*ptr   = L'.';
-				*++ptr = bup_setting.GetBackupExtChar();
-				*++ptr = L'0';
-				*++ptr = L'0';
-				*++ptr = L'\0';
+				// _MAX_EXT = 256なので桁溢れは考慮しない。
+				const WCHAR tmp[] = { L'.', bup_setting.GetBackupExtChar(), L'0', L'0', 0 };
+				auto_snprintf_s(ptr, _countof(tmp), L"%s", tmp);
```

Comment:
```suggestion
				auto_snprintf_s(ptr, _MAX_EXT - (ptr - szExt), L"%s", tmp);
```

ってことじゃないですかね？

中間変数 ptr が複雑な構造を生み出しているように思います。
構造を取っ払ってしまえば見やすくなりますが、その修正ではレビュー不能という判断なんでしょうか。

```c++
    if( bup_setting.GetBackupType() == 3 ){
        ::wcscpy_s(szExt, L"");
    }
    const WCHAR tmp[] = { L'.', bup_setting.GetBackupExtChar(), L'0', L'0', 0 };
    ::wcscat_s(szExt, tmp);
```

この件も似たのがないとは考えにくいので、まとめて対応してはどうかと思います。

Reply:
調べてみたんですが、似たパターンはないようでした。